packages/ui/src/components/va-data-table/hooks/useTableScroll.ts

```diff
@@ -0,0 +1,35 @@
+interface useTableScrollProps {
+  stickyHeader: boolean
+  stickyFooter: boolean
+  height: string | number | undefined
+  scrollEventsThreshold: number | undefined
+}
+
+export default function useTableScroll (
+  props: useTableScrollProps,
+  emit: (event: 'scroll:bottom' | 'scroll:top', ...args: any[]) => void,
+) {
+  const onScroll = (evt: Event) => {
+    if (!(props.stickyHeader || props.stickyFooter || props.height)) {
+      return
+    }
+
+    if (props.scrollEventsThreshold === undefined) {
+      return
+    }
+
+    const { scrollTop, scrollHeight, clientHeight } = evt.target as HTMLTableElement
+
+    if (scrollHeight - clientHeight - scrollTop <= props.scrollEventsThreshold) {
```

Comment:
I think `scrollEventsThreshold` should be renamed to "margin"... Or not?
If this made for lazy-loading I suggest following [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)

Reply:
reworked with [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)